bindgen/ir/comp.rs

```diff
@@ -643,8 +653,15 @@ where
             // NB: The `bitfield_width` here is completely, absolutely
             // intentional.  Alignment of the allocation unit is based on the
             // maximum bitfield width, not (directly) on the bitfields' types'
-            // alignment.
-            unit_align = cmp::max(unit_align, bitfield_width);
+            // alignment. this is only available in bitfields start of structure
+            if bitfields_bit_begin_off == 0 {
+                unit_align = cmp::max(unit_align, bitfield_width);
+            } else {
+                // if bitfields is not start of structure, due to the presence
```

Comment:
Something like:

```
If the bitfield is not the first member of the struct, we can't use max alignment. Default to u8.
```

What is `backward embedding`? Never heard that term. You mean something like the optimization where you put fields in the padding of the base class or something?